src/cmd/pkg/pagination/list.go

```diff
@@ -0,0 +1,44 @@
+package pagination
+
+// Page is a list page.
+type Page struct {
+	Size  int64
+	Token string
+}
+
+// List invokes the listFn repeatedly accumulating results per page until p.Size is met or results are exhausted.
+func List[T any](p Page,
+	listFn func(p Page) (items []T, token string, err error),
+) ([]T, string, error) {
+	var (
+		items     []T
+		pageSize  = p.Size
+		nextToken = p.Token
+	)
+
+	// loop until we've fetched enough results
+	for {
+		current, token, err := listFn(Page{Size: pageSize, Token: nextToken})
```

Comment:
based on what im reading, if i try to send in pageSize == 1 million it will attempt to get 1 million records in a go.  does this fail or default to the max allowed in API?

Reply:
It will not fail in the API. The max amount of results we see in practice is <20k in most tenants. This API returns about 100k max results for our largest tenant.

Reply:
Yeah the BE should be enforcing some max so if PageSize is > that max, or if PageSize == 0, that max is what is used.